src/modules/auth/auth.service.ts

```diff
@@ -1,55 +1,232 @@
+      },
+      accessToken: tokens.accessToken,
+      refreshToken: tokens.refreshToken,
+    };
+  }
+
+  /**
+   * Refresh access and refresh tokens
+   * @param refreshToken - Current refresh token
+   * @returns New access and refresh tokens
+   * @throws UnauthorizedException if token is invalid or expired
+   */
+  public async refresh(
+    refreshToken: string,
+  ): Promise<{ accessToken: string; refreshToken: string }> {
     try {
-      const { identifiers } = await this.userRepo
-        .createQueryBuilder()
-        .insert()
-        .into(UserEntity)
-        .values(dto)
-        .execute();
-
-      const insertedId = identifiers?.[0]?.id as string;
-      if (!insertedId) {
-        throw new InternalServerErrorException('Failed to create user');
+      // Verify and decode refresh token with explicit type
+      const payload = this.jwtService.verify<JwtPayload>(refreshToken, {
+        secret: this.configService.get<string>('JWT_REFRESH_SECRET'),
+      });
+
+      // Find user by ID from token payload
+      const user = await this.userRepo.findOne({
+        where: { id: payload.sub },
+      });
+
+      if (!user || !user.refreshToken) {
+        throw new UnauthorizedException('Invalid refresh token');
       }
 
-      const created = await this.userRepo.findOneBy({ id: insertedId });
-      if (!created) {
-        throw new InternalServerErrorException('Created user not found');
+      if (user.refreshToken !== refreshToken) {
+        throw new UnauthorizedException('Invalid refresh token');
       }
 
-      return UserResponseDto.fromEntity(created);
-    } catch (error) {
-      this.logger.error('Register failed', (error as Error).message);
-      throw error;
+      // Generate new tokens
+      const tokens = await this.generateTokens(user);
+
+      // Update refresh token in database
+      await this.userRepo.update(user.id, {
+        refreshToken: tokens.refreshToken,
+      });
+
+      return {
+        accessToken: tokens.accessToken,
+        refreshToken: tokens.refreshToken,
+      };
+    } catch {
+      // Catch block without unused variable
+      throw new UnauthorizedException('Invalid or expired refresh token');
     }
   }
 
-  public async findAll(): Promise<UserResponseDto[]> {
-    const users = await this.userRepo
-      .createQueryBuilder('u')
-      .select(['u.id', 'u.username'])
-      .getMany();
+  /**
+   * Logout user
+   * Removes refresh token from database
+   * @param userId - User ID to logout
+   * @returns Success message
+   */
+  public async logout(userId: string): Promise<{ message: string }> {
+    await this.userRepo.update(userId, { refreshToken: null });
+    return { message: 'Logged out successfully' };
+  }
+
+  /**
+   * Get user profile by ID
+   * @param userId - User ID
+   * @returns User profile data
+   * @throws UnauthorizedException if user not found
+   */
+  public async getProfile(userId: string): Promise<UserResponseDto> {
```

Comment:
but we will not have a profile. We only need to gather the email of the users who use service